src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -32,39 +34,59 @@ func (client *Client) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 // connetion and starts piping the data over a new net.Conn obtained from the
 // given dial function.
 func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
-	if req.Method != "CONNECT" {
+	var err error
+
+	// intercept can only by used for CONNECT requests.
+	if req.Method != httpConnectMethod {
 		panic("Intercept used for non-CONNECT request!")
 	}
 
-	// Hijack underlying connection
-	clientConn, _, err := resp.(http.Hijacker).Hijack()
-	if err != nil {
+	// Hijacking underlying connection.
+	var clientConn net.Conn
+	if clientConn, _, err = resp.(http.Hijacker).Hijack(); err != nil {
```

Comment:
Yeah, I understand the pattern.  What I meant is I think I read somewhere that Go would allow you to say `clientConn, _, err := ...`, that is, use `:=` even though `err` has already been declared before.  So you could use this idiom without having to declare `clientConn` before.

But I tried this in a test program and I was wrong.  If you do this, `clientConn` would be scoped to the body of the `if`.  I must have confused this with some other special case involving errors.